src/razor/src/document/razorDocumentManager.ts

```diff
@@ -193,11 +193,13 @@ export class RazorDocumentManager implements IRazorDocumentManager {
     private closeDocument(uri: vscode.Uri) {
         const document = this._getDocument(uri);
 
-        // Files outside of the workspace will return undefined from getWorkspaceFolder
-        const workspaceFolder = vscode.workspace.getWorkspaceFolder(uri);
-        if (!workspaceFolder) {
-            // Out of workspace files should be removed once they're closed
-            this.removeDocument(uri);
+        // Documents that are files should be removed if they are outside the workspace folder
+        if (uri.scheme === 'file') {
```

Comment:
it depends - do you need to close non file documents?  Other non-file documents that I know of
1.  Documents just created, but not saved
2.  Diff window files

Reply:
Diff window files should also not cause us to remove. It looks like we don't track documents that are just created but not saved.